app/main.py

```diff
@@ -1,34 +1,143 @@
+import dataclasses
+from typing import Union
+
+
+class ErrorTotalNumberShips(Exception):
+    pass
+
+
+class ErrorNumberDecks(Exception):
+    pass
+
+
+class ErrorNeighboringCells(Exception):
+    pass
+
+
+class ErrorNotOneLineShips(Exception):
+    pass
+
+
+@dataclasses.dataclass
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    row: int
+    column: int
+    is_alive: bool = True
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(
+            self,
+            start: tuple,
+            end: tuple,
+            is_drowned: bool = False) -> None:
```

Comment:
fix the breaking line here - move closing quote to the next line or self to the open quote